docs/network-spec/miniprotocols.tex

```diff
@@ -1382,6 +1382,41 @@ \subsection{Client Implementation Details}
 
 If a peer gets disconnected, it should get unregistered.
 
+\subsubsection{Deciding to whom to request peers to (and how many)}
+
+First of all peer sharing requests should only be issued if:
+
+\begin{itemize}
+  \item The current number of known peers is less than the target for known peers;
+  \item The rate limit value for peer sharing requests isn't exceeded;
+  \item There are available peers to issue requests too;
+\end{itemize}
+
+If these conditions hold then we can pick a set of peers to issue requests to.
+Ideally this set respects the rate limit value for peer sharing requests.
+
+If a peer has \texttt{PeerSharingDisabled} flag value do not ask it for peers.
+This peer won't even have the Peer Sharing miniprotocol server running.
+
+The amount of peers to request to each upstream peer should aim to fullfill
+the target for known peers. This number should be split for the current peer
+target objective across all peer sharing candidates for efficiency and
+diversity reasons.
+
+\subsubsection{Picking peers from the response}
+
+Apart from managing the Outbound Governor state correctly, the final result
+set should be a random distribution of the original set.
+
+This selection should be done in such a way that when the same initial PRNG
+state is used, the selected set does not significantly vary with small
+perturbations in the set of published peers.
```

Comment:
> published peers

Do you mean the received peers through peer sharing, right?

Reply:
Yes